TeamCode/src/main/java/com/acmerobotics/relicrecovery/motion/MotionProfileGenerator.java

```diff
@@ -159,11 +159,67 @@ else if(constraints.endBehavior == MotionConstraints.EndBehavior.VIOLATE_MAX_ABS
         profile.appendControl(0, timesToDecel[1]);
         profile.appendControl(j, timesToDecel[2]);
 
-        profile.doCommenceDeUnConsolidatedification();
+        profile.removeEmptySegments();
 
         return profile;
     }
 
+    public static MotionProfile generateStoppingProfile(MotionState start, MotionConstraints constraints) {
+        double dv = -start.v;
+
+        if (dv < 0) {
+            MotionProfile profile = generateStoppingProfile(start.flipped(), constraints);
+            SuperArrayList<MotionSegment> segments = profile.segments();
+            //then flip the result
+            SuperArrayList<MotionSegment> flipped = new SuperArrayList<>();
+            for (MotionSegment seg: segments) {
+                flipped.add(new MotionSegment(seg.start().flipped(), seg.dt()));
+            }
+            return new MotionProfile (flipped, constraints);
+        }
+
+        MotionProfile profile = new MotionProfile(start, constraints);
+
+        if (start.a < 0) {
+            double stopingTime = Math.abs(start.a / constraints.maxJ);
+            profile.appendControl(constraints.maxJ, stopingTime);
+            start = profile.end();
+            dv = -start.v;
+        }
+
+        double aMax = Math.min(constraints.maxA, Math.sqrt((start.a * start.a / 2) + (dv * constraints.maxJ)));
+        System.out.println(Math.sqrt((start.a * start.a / 2) + (dv * constraints.maxJ)));
```

Comment:
Can I get rid of these print statements?

Reply:
done